src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_attend_and_excite.py

```diff
@@ -0,0 +1,964 @@
+import abc
+import math
+import numbers
+from typing import Any, Callable, Dict, List, Optional, Tuple, Union
+
+import numpy as np
+import torch
+from torch.nn import functional as F
+
+import cv2
+from PIL import Image
+from transformers import CLIPFeatureExtractor, CLIPTextModel, CLIPTokenizer
+
+from ...models import AutoencoderKL, UNet2DConditionModel
+from ...models.cross_attention import CrossAttention
+from ...schedulers import KarrasDiffusionSchedulers
+from ...utils import logging
+from . import StableDiffusionPipeline, StableDiffusionPipelineOutput
+from .safety_checker import StableDiffusionSafetyChecker
+
+
+logger = logging.get_logger(__name__)
+
+EXAMPLE_DOC_STRING = """
+    Examples:
+        ```py
+        >>> import torch
+        >>> from diffusers import StableDiffusionAttendAndExcitePipeline
+
+        >>> pipe = StableDiffusionAttendAndExcitePipeline.from_pretrained(
+        ...     "CompVis/stable-diffusion-v1-4", torch_dtype=torch.float16
+        ... ).to("cuda")
+
+        >>> pipe = pipe.to("cuda")
+
+        >>> prompt = "a cat and a frog"
+        >>> indices_to_alter = [2, 5]
+        >>> is_attend_and_excite = True
+        >>> seed = 6141
+        >>> g = torch.Generator("cuda").manual_seed(seed)
+
+        >>> images = pipe(
+        ...     prompt=[prompt],
+        ...     indices_to_alter=indices_to_alter,
+        ...     attention_res=16,
+        ...     guidance_scale=7.5,
+        ...     generator=g,
+        ...     num_inference_steps=50,
+        ...     max_iter_to_alter=25,
+        ...     run_standard_sd= not is_attend_and_excite,
```

Comment:
would maybe try to use less costumized inputs in the examples, maybe just something like:

```python
 >>> images = pipe(
        ...     prompt=[prompt],
        ...     indices_to_alter=indices_to_alter,
        ...     generator=g,
        ...     num_inference_steps=50,
        ...     max_iter_to_alter=25,
        ...     ).images
```

For all others the defaults should be good :-)